command/ssh.go

```diff
@@ -0,0 +1,180 @@
+package command
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/builtin/logical/ssh"
+)
+
+// SSHCommand is a Command that establishes a SSH connection
+// with target by generating a dynamic key
+type SSHCommand struct {
+	Meta
+}
+
+func (c *SSHCommand) Run(args []string) int {
+	var role string
+	var port string
+	var sshCmdArgs []string
+	var sshDynamicKeyFileName string
+	flags := c.Meta.FlagSet("ssh", FlagSetDefault)
+	flags.StringVar(&role, "role", "", "")
+	flags.StringVar(&port, "port", "22", "")
+	flags.Usage = func() { c.Ui.Error(c.Help()) }
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+	args = flags.Args()
+	if len(args) < 1 {
+		c.Ui.Error("ssh expects at least one argument")
+		return 2
+	}
+
+	client, err := c.Client()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		return 2
+	}
+
+	input := strings.Split(args[0], "@")
+	if len(input) != 2 {
```

Comment:
I agree with using user.Current() instead of default_user.
1) User of the service may not be aware of 'default_user' being chosen in place of current user, and expect otherwise.
2) If default_user needs to be used, there will be an overhead of reading role.

Going ahead with user.Current() unless you object.